File: artifacts/tulipindicators/manual_translation/src/crossover.rs
use crate::*;

pub fn ti_crossover_start(options: Option<&[f64]>) -> i32 {
	println!("\n* ti_crossover_start");
    let _ = options; // Explicitly ignore the parameter to match C behavior
    1
}
pub fn ti_crossover(
    size: i32,
    inputs: Option<&[&[f64]]>,
    options: Option<&[f64]>,
    outputs: Option<&mut [&mut [f64]]>,
) -> i32 {
	println!("\n* ti_crossover");
    let inputs = inputs.unwrap();
    let outputs = outputs.unwrap();
    
    if inputs.len() < 2 {
        return 0;
    }
    let a = inputs[0];
    let b = inputs[1];
    let _ = options; // explicitly ignore options to match C behavior
    let output = &mut outputs[0];
    let mut output_idx = 0;

    for i in 1..size as usize {
        output[output_idx] = if (a[i] > b[i]) && (a[i - 1] <= b[i - 1]) {
            1.0
        } else {
            0.0
        };
        output_idx += 1;
    }

    0
}

File: artifacts/tulipindicators/decompose/src/crossover.c
int ti_crossover_start(const double *options)
{
  (void) options;
  return 1;
}


/*** DEPENDENCIES:
***/


int ti_crossover(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *a = inputs[0];
  const double *b = inputs[1];
  (void) options;
  double *output = outputs[0];
  unsigned int output_idx = 0;
  int i;
  for (i = 1; i < size; i += 1)
  {
    output[output_idx] = (a[i] > b[i]) && (a[i - 1] <= b[i - 1]);
    output_idx += 1;
  }

  return 0;
}


/*** DEPENDENCIES:
***/


